File: Farhan sir/Arrays/gameEntry.py
class Player:
    def __init__(self, name, score):
        self.name = name
        self.score = score

class Array:
    def __init__(self, capacity):
        self.array = [None] * capacity
        self.size = 0

    def add(self, entry):
        new_score = entry.score

        if self.size < len(self.array):
            self.array[self.size] = entry
            self.size += 1
        elif new_score > self.array[self.size - 1].score:
            self.array[self.size - 1] = entry

        for i in range(self.size - 1):
            for j in range(self.size - 1 - i):
                if self.array[j].score < self.array[j + 1].score:
                    self.array[j], self.array[j + 1] = self.array[j + 1], self.array[j]

    def remove(self, i):
        if i < 0 or i >= self.size:
            raise IndexError(f"Invalid index: {i}")
        temp = self.array[i]

        for j in range(i, self.size - 1):
            self.array[j] = self.array[j + 1]
        self.array[self.size - 1] = None
        self.size -= 1
        return temp

    def show(self):
        for i in range(self.size):
            print(f"{i + 1}. Name: {self.array[i].name}, Score: {self.array[i].score}")
        print()


def main():
    scoreboard = Array(3)
    print("Game Start")

    while True:
        entry = input("Enter name and score (enter 'over' to stop the game): ")
        if entry == "over": break
        name, score = entry.split()
        score = int(score)
        game_entry = Player(name, score)
        scoreboard.add(game_entry)
        print("Current scoreboard")
        scoreboard.show()

    print("Game over")

if __name__ == "__main__":
    main()

File: Farhan sir/Arrays/gameEntry.java
import java.util.*;

class Player 
{
    private final String name;
    private final int score;

    public Player(String name, int score) 
    {
        this.name = name;
        this.score = score;
    }

    public String getName() 
    {
        return name;
    }

    public int getScore() 
    {
        return score;
    }

    @Override
    public String toString() 
    {
        return this.name + ", " + this.score;
    }
}

class Array 
{
    private final Player array[];
    private int size;

    public Array(int capacity) 
    {
        array = new Player[capacity];
        size = 0;
    }

    public void add(Player entry) 
    {
        int newScore = entry.getScore();

        if (size < array.length) 
        {
            array[size] = entry;
            size++;
        } 
        else if (newScore > array[size - 1].getScore()) 
        {
            array[size - 1] = entry;
        }

        for (int i = 0; i < size - 1; i++) 
        {
            for (int j = 0; j < size - 1 - i; j++) 
            {
                if (array[j].getScore() < array[j + 1].getScore()) 
                {
                    Player temp = array[j];
                    array[j] = array[j + 1];
                    array[j + 1] = temp;
                }
            }
        }
    }

    public Player remove(int i) 
    {
        if (i < 0 || i >= size) 
        {
            throw new IndexOutOfBoundsException("Invalid index: " + i);
        }

        Player temp = array[i];

        for (int j = i; j < size - 1; j++) 
        {
            array[j] = array[j + 1];
        }
        array[size - 1] = null;
        size--;

        return temp;
    }

    public void show() 
    {
        for (int i = 0; i < size; i++) 
        {
            System.out.println((i + 1) + ". Name: " + array[i].getName() + ", Score: " + array[i].getScore());
        }
        System.out.println();
    }
}

public class gameEntry 
{
    public static void main(String[] args) {
        Array scoreboard = new Array(3);
        Scanner scanner = new Scanner(System.in);

        System.out.println("Game Start");

        while (true) {
            System.out.print("Enter name and score (enter 'over' to stop the game): ");
            String entry = scanner.nextLine();

            if (entry.equals("over"))
                break;

            String[] parts = entry.split(" ");
            String name = parts[0];
            int score = Integer.parseInt(parts[1]);
            Player gameEntry = new Player(name, score);
            scoreboard.add(gameEntry);

            System.out.println("Current scoreboard");
            scoreboard.show();
        }

        System.out.println("Game over");
        scanner.close();
    }
}
